TYP: unexpected `var-annotated` error with masked array

### Describe the issue:

`mypy` raises `var-annotated` on `x1`, but `x2` seems fine.

- x-ref: https://github.com/python/typeshed/pull/15472#issuecomment-4470315371
- cc @jorenham found this while debugging the `list.{count, index, remove}` PR in typeshed
- This may possibly be related to mypy's long standing issue with bidirectional inference (https://github.com/python/mypy/issues/19304)


### Reproduce the code example:

```python
import numpy as np

y: list[float] = [-1, -1, 10, 10.1, 10.2]
x1 = np.ma.array(y, mask=[True, True, False, False, False])
x2 = np.ma.array([-1, -1, 10, 10.1, 10.2], mask=[True, True, False, False, False])
```

### Error message:

```shell
tmp.py:4: error: Need type annotation for "x1"  [var-annotated]
Found 1 error in 1 file (checked 1 source file)
```

### Python and NumPy Versions:

- Python 3.14.5
- numpy               2.4.5

### Type-checker version and settings:

- mypy 2.2.0+dev.11008008752aa10000a6230322fc4ebe3eb2b456 (compiled: no)
- mypy 2.1.0 (compiled: yes)

### Additional typing packages.

_No response_

--- a/numpy/ma/core.pyi
+++ b/numpy/ma/core.pyi
@@ -2874,7 +2874,7 @@ def array[ScalarT: np.generic](
     ndmin: int = 0,
 ) -> _MaskedArray[ScalarT]: ...
 @overload
-def array[ScalarT: np.generic](
+def array(
     data: object,
     dtype: DTypeLike | None = None,
     copy: bool = False,
@@ -2886,7 +2886,7 @@ def array[ScalarT: np.generic](
     shrink: bool = True,
     subok: bool = True,
     ndmin: int = 0,
-) -> _MaskedArray[ScalarT]: ...
+) -> _MaskedArray[Any]: ...
 
 # keep in sync with `array`
 @overload
@@ -2902,11 +2902,11 @@ def asarray[ScalarT: np.generic](
     order: _OrderKACF | None = None,
 ) -> _MaskedArray[ScalarT]: ...
 @overload
-def asarray[ScalarT: np.generic](
+def asarray(
     a: object,
     dtype: DTypeLike | None = None,
     order: _OrderKACF | None = None,
-) -> _MaskedArray[ScalarT]: ...
+) -> _MaskedArray[Any]: ...
 
 # keep in sync with `asarray` (but note the additional first overload)
 @overload
@@ -2924,11 +2924,11 @@ def asanyarray[ScalarT: np.generic](
     order: _OrderKACF | None = None,
 ) -> _MaskedArray[ScalarT]: ...
 @overload
-def asanyarray[ScalarT: np.generic](
+def asanyarray(
     a: object,
     dtype: DTypeLike | None = None,
     order: _OrderKACF | None = None,
-) -> _MaskedArray[ScalarT]: ...
+) -> _MaskedArray[Any]: ...
 
 #
 def is_masked(x: object) -> bool: ...

